@here/harp-mapview/lib/Utils.ts

```diff
@@ -242,53 +243,194 @@ export namespace MapViewUtils {
+        // Note, the points Rp, C, and M create a plane that is perpendicular to the earths surface,
+        // because the tilt axis is perpendicular to the up vector.
+        // The following variable `angleBetweenRotationTargetAndMapTarget` is the angle between the
+        // two rays C->Rp and C->M, i.e. the angle RpCM. This angle remains constant when tilting with
+        // a fixed `offsetX` and `offsetY`. It is calculated by using the intersection of the two
+        // ways with the earth. Where:
+        // MRpC is 90 + `angleBetweenNormals` - tilt at Rp.
+        // CMRp is 90 + `tilt`
+        // RpCM is 180 - CRpM - RpMC, aka `angleBetweenRotationTargetAndMapTarget`
+        // Note the use of `angleBetweenNormals` to ensure this works for spherical projections.
+        // Note, also, the above calculation only works when the tilt at the M is less than the tilt
+        // at Rp, otherwise the above formula won't work. We however don't need to worry about this
+        // case because this happens only when offsetY is less than zero, and this is handled above.
+        const angleBetweenRotationTargetAndMapTarget =
+            ninetyRad * 2 -
+            (ninetyRad + angleBetweenNormals - rotationCenterTilt + ninetyRad + tilt);
+
+        // This uses the same formula as above, just we change `tilt` to the maxTiltAngle,
+        // i.e we want to find the greatest angle at the rotation target that gives us the max
+        // angle at the map center target.
+        // i.e. CMR is 90 + `maxTiltAngle`
+        const maxRotationTargetAngle =
+            ninetyRad * 2 - angleBetweenRotationTargetAndMapTarget - ninetyRad - maxTiltAngle;
+
+        // Converting the `maxRotationTargetAngle` back to a tilt is as easy as subtracting it from
+        // 90 and the `angleBetweenNormals`, i.e. this gives us the maximum allowed tilt at R that
+        // satisfies the `maxTiltAngle` constraint. Note, for globe projection, this is just an
+        // approximation, because once we move the camera by delta, the map target changes, and
+        // therefore the normal also changes, this would need to be applied iteratively until the
+        // difference in normals is reduced to some epsilon. I don't apply this because it is
+        // computationally expensive and the user would never notice this in practice.
+        const maxTilt = ninetyRad + angleBetweenNormals - maxRotationTargetAngle;
+
+        // Here we clamp to the min of `maxTilt` and 89 degrees. The check for 89 is to prevent it
+        // intersecting with the world at 90. This is possible for example when the R position is
+        // near the horizon. If the angle RCM is say 5 degrees, then an angle of say 89 degrees at
+        // R, plus 5 degrees means the tilt at M would be 84 degrees. If the `maxTiltAngle` is
+        // however 89, then we still have 5 degrees at M, so, without this check, we would go to 90
+        // degrees.
```

Comment:
Could you try making these comments a little bit more concise?

Reply:
Please highlight which part of the text is superfluous.

Reply:
nevermind, it's ok.

Reply:
I made some changes anyway with the changes you mentioned, so it should be a bit better now.